onnxruntime/core/mlas/lib/dgemm.cpp

```diff
@@ -885,121 +845,42 @@ Return Value:
     // works okay for operations involving skinny matrices.
     //
 
+    ptrdiff_t ThreadsPerGemm =
```

Comment:
I'm wondering if it always gets better performance by partitioning like this. For example, if batch size is 4 and TargetThreadCount is 6, then only 4 threads are used. By contrast, simply looping over the batch will use all 6 threads.